File: LeetCode/Problems/66.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        digits.insert(0, 0)
        def increment(i):
            if digits[i] == 9:
                digits[i] = 0
                increment(i-1)
            else:
                digits[i] += 1

        increment(len(digits) - 1)

        if digits[0] == 0:
            digits.pop(0)
        return digits

File: leetcode/66.cpp
class Solution
{
private:
    vector<int> digits;
    void increment(int i)
    {
        digits[i]++;
        if (digits[i] == 10)
        {
            digits[i] = 0;
            if (i <= 0)
            {
                digits.insert(digits.begin(), 1);
            }
            else
            {
                increment(i - 1);
            }
        }
    }

public:
    vector<int> plusOne(vector<int> &digits)
    {
        this->digits = digits;
        increment(digits.size() - 1);
        return this->digits;
    }
};
